src/hotspot/os_cpu/linux_s390/os_linux_s390.cpp

```diff
@@ -460,20 +460,25 @@ void os::print_tos_pc(outputStream *st, const void *context) {
   st->cr();
 }
 
-void os::print_register_info(outputStream *st, const void *context) {
-  if (context == nullptr) return;
+void os::print_register_info(outputStream *st, const void *context, int& continuation) {
+  const int register_count = 16 /* r0-r15 */ + 1 /* pc */;
+  int n = continuation;
+  if (context == nullptr || n < 0 || n >= register_count) {
+    return;
+  }
 
   const ucontext_t *uc = (const ucontext_t*)context;
-
-  st->print_cr("Register to memory mapping:");
-  st->cr();
-
-  st->print("pc ="); print_location(st, (intptr_t)uc->uc_mcontext.psw.addr);
-  for (int i = 0; i < 16; i++) {
-    st->print("r%-2d=", i);
-    print_location(st, uc->uc_mcontext.gregs[i]);
+  while (n < register_count) {
+    // Update continuation with next index before printing location
+    continuation = n + 1;
+    if (n == 0) {
+      st->print("pc ="); print_location(st, (intptr_t)uc->uc_mcontext.psw.addr);
+    } else {
+      st->print("r%-2d=", n-1);
+      print_location(st, uc->uc_mcontext.gregs[n-1]);
```

Comment:
The "-1" here and the "-3" for ppc makes me a bit nervous, since if this bitrots, we probably never notice that we print the wrong registers unless someone debugs. Pragmatic proposal: Print the general regs first, then the special ones, then you don't need to deal with hard-coded offsets. I don't think anyone would object to that.